controllers/user/UserController.php

```diff
@@ -110,10 +112,48 @@ public function userLogin($username,$password){
 
         $jwt = JWT::encode($data, $config->jwtKey, 'HS512');
 
+        //Accept the privacy policy page
+        $this->acceptPrivacy($user);
+
         return $this->returnSuccess('Success', 200, [
             'auth_token' => $jwt,
             'expired_at' => (!isset($data['exp'])) ? 0 : $data['exp']
         ]);
 
     }
+
+    /**
+     * Accept the privacy policy, After the login
+     * We are sending dataPrivacyCheck key with status(true/false), if it is true only then accept the privacy policy
+     *
+     * @param $user
+     * @return array|bool|void
+     */
+    public function acceptPrivacy($user){
+        //Accept the privacy policy
+        $model = new RegistrationChecks();
+        $model->load(Yii::$app->request->getBodyParam("legal", []), '');
+
+        //Check the dataPrivacyCheck key is true/false
+        if($model->dataPrivacyCheck){
```

Comment:
where is checked, that `dataPrivacyCheck` is `true` when passing

```
"legal": {
  "dataPrivacyCheck": true
}
```

and not `false`?
isn't this just checking if the key `dataPrivacyCheck` exists and it does not matter if `true` or `false`? or did i get something wrong?

Reply:
i could achieve registrations with `true`, `false` and even without `legal` params

Reply:
Hi @donni106 
Here we are checking dataPrivacyCheck is true or false
![image](https://user-images.githubusercontent.com/103625795/169637458-067174bd-7866-4113-a3c9-e088a6ab49ab.png)

The value of dataPrivacyCheck should be 1/true because it is mentioned in rules of RegistrationCheck model.
![image](https://user-images.githubusercontent.com/103625795/169637568-d092a887-2ad7-462c-b00a-aa996c1318a5.png)

`isn't this just checking if the key dataPrivacyCheck exists now and it does not matter if true or false? or did i get something wrong?`
So, I don't think only dataPrivacyCheck key will make sense.


`i could achieve registrations with true, false and even without legal params. what could be missing?`
Yes, you can
If the dataPrivacyCheck value will be true and user will automatically accept privacy polic condition and if it is false then user will not accept privacy polic during registration.

And if we remove legal params from body then user will be registered but without accepting privacy policy condition.

Reply:
can we avoid the registration to be successful in the case that `legal` key is missing or `dataPrivacyCheck` is `false`?

Reply:
Hi @donni106 
Yes, we can but what if the legal module is not activated/disabled?

Reply:
you made that update about checking if the legal module exists. so only if this is the case, registration should rely on `dataPrivacyCheck` and block if it is missing or false. otherwise, without legal module, it should behave as before and be successful without. can we achieve it in this way?